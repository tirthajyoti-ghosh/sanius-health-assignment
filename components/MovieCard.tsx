import React from 'react';
import { StyleSheet, Image, TouchableOpacity, View, Platform } from 'react-native';
import { RelativePathString, useRouter } from 'expo-router';
import { ThemedText } from '@/components/ThemedText';
import { ThemedView } from '@/components/ThemedView';
import { Movie } from '@/api/types';
import { getImageUrl } from '@/api/movie';
import { formatDate } from '@/utils/date';

type MovieCardProps = {
  movie: Movie;
};

export function MovieCard({ movie }: MovieCardProps) {
  const router = useRouter();
  
  const handlePress = () => {
    router.push(`/movie/${movie.id.toString()}` as RelativePathString);
  };

  return (
    <TouchableOpacity 
      style={styles.container}
      onPress={handlePress}
      activeOpacity={0.7}
    >
      <ThemedView style={styles.card}>
        <View style={styles.imageContainer}>
          {movie.poster_path ? (
            <Image
              source={{ uri: getImageUrl(movie.poster_path, 'w342') || '' }}
              style={styles.poster}
              resizeMode="cover"
            />
          ) : (
            <View style={styles.noImage}>
              <ThemedText>No Image</ThemedText>
            </View>
          )}
          <View style={styles.ratingContainer}>
            <ThemedView style={styles.ratingBadge}>
              <ThemedText style={styles.ratingText}>{movie.vote_average.toFixed(1)}</ThemedText>
            </ThemedView>
          </View>
        </View>
        
        <View style={styles.infoContainer}>
          <ThemedText type="defaultSemiBold" numberOfLines={2}>{movie.title}</ThemedText>
          <ThemedText style={styles.date}>{formatDate(movie.release_date)}</ThemedText>
        </View>
      </ThemedView>
    </TouchableOpacity>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    margin: 8,
  },
  card: {
    borderRadius: 12,
    overflow: 'hidden',
    ...Platform.select({
      ios: {
        shadowColor: '#000',
        shadowOffset: { width: 0, height: 2 },
        shadowOpacity: 0.1,
        shadowRadius: 4,
      },
      android: {
        elevation: 3,
      },
    }),
  },
  imageContainer: {
    position: 'relative',
  },
  poster: {
    width: '100%',
    aspectRatio: 2/3,
  },
  noImage: {
    width: '100%',
    aspectRatio: 2/3,
    backgroundColor: '#d1d1d1',
    justifyContent: 'center',
    alignItems: 'center',
  },
  ratingContainer: {
    position: 'absolute',
    bottom: 8,
    right: 8,
  },
  ratingBadge: {
    backgroundColor: '#0a7ea4',
    paddingHorizontal: 8,
    paddingVertical: 4,
    borderRadius: 16,
  },
  ratingText: {
    color: 'white',
    fontWeight: 'bold',
    fontSize: 12,
  },
  infoContainer: {
    padding: 12,
    gap: 4,
  },
  date: {
    fontSize: 14,
    opacity: 0.7,
  },
});