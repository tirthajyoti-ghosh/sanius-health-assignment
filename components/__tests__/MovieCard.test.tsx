import * as React from 'react';
import {render} from '@testing-library/react-native';
import { MovieCard } from '../MovieCard';

jest.mock('expo-router', () => ({
  useRouter: () => ({
    push: jest.fn(),
  }),
}));

jest.mock('@/api/movie', () => ({
  getImageUrl: (path: string) => path ? `https://image.tmdb.org/t/p/w342${path}` : null,
}));

const mockMovie = {
  id: 1,
  title: 'Test Movie',
  poster_path: '/test-path.jpg',
  backdrop_path: null,
  release_date: '2023-01-01',
  vote_average: 7.5,
  overview: 'Test overview',
  genre_ids: [1, 2],
  adult: false,
  original_language: 'en',
  original_title: 'Test Movie Original',
  popularity: 123.45,
  video: false,
  vote_count: 1000,
};

it(`renders movie card correctly`, () => {
  const tree = render(<MovieCard movie={mockMovie} />).toJSON();
  expect(tree).toMatchSnapshot();
});

it(`renders movie card without poster correctly`, () => {
  const movieWithoutPoster = { ...mockMovie, poster_path: null };
  const tree = render(<MovieCard movie={movieWithoutPoster} />).toJSON();
  expect(tree).toMatchSnapshot();
});