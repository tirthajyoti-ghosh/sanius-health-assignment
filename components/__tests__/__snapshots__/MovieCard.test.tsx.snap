// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders movie card correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "flex": 1,
      "margin": 8,
      "opacity": 1,
    }
  }
>
  <View
    style={
      [
        {
          "backgroundColor": "#fff",
        },
        {
          "borderRadius": 12,
          "overflow": "hidden",
          "shadowColor": "#000",
          "shadowOffset": {
            "height": 2,
            "width": 0,
          },
          "shadowOpacity": 0.1,
          "shadowRadius": 4,
        },
      ]
    }
  >
    <View
      style={
        {
          "position": "relative",
        }
      }
    >
      <Image
        resizeMode="cover"
        source={
          {
            "uri": "https://image.tmdb.org/t/p/w342/test-path.jpg",
          }
        }
        style={
          {
            "aspectRatio": 0.6666666666666666,
            "width": "100%",
          }
        }
      />
      <View
        style={
          {
            "bottom": 8,
            "position": "absolute",
            "right": 8,
          }
        }
      >
        <View
          style={
            [
              {
                "backgroundColor": "#fff",
              },
              {
                "backgroundColor": "#0a7ea4",
                "borderRadius": 16,
                "paddingHorizontal": 8,
                "paddingVertical": 4,
              },
            ]
          }
        >
          <Text
            style={
              [
                {
                  "color": "#11181C",
                },
                {
                  "fontSize": 16,
                  "lineHeight": 24,
                },
                undefined,
                undefined,
                undefined,
                undefined,
                {
                  "color": "white",
                  "fontSize": 12,
                  "fontWeight": "bold",
                },
              ]
            }
          >
            7.5
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        {
          "gap": 4,
          "padding": 12,
        }
      }
    >
      <Text
        numberOfLines={2}
        style={
          [
            {
              "color": "#11181C",
            },
            undefined,
            undefined,
            {
              "fontSize": 16,
              "fontWeight": "600",
              "lineHeight": 24,
            },
            undefined,
            undefined,
            undefined,
          ]
        }
      >
        Test Movie
      </Text>
      <Text
        style={
          [
            {
              "color": "#11181C",
            },
            {
              "fontSize": 16,
              "lineHeight": 24,
            },
            undefined,
            undefined,
            undefined,
            undefined,
            {
              "fontSize": 14,
              "opacity": 0.7,
            },
          ]
        }
      >
        January 1, 2023
      </Text>
    </View>
  </View>
</View>
`;

exports[`renders movie card without poster correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "flex": 1,
      "margin": 8,
      "opacity": 1,
    }
  }
>
  <View
    style={
      [
        {
          "backgroundColor": "#fff",
        },
        {
          "borderRadius": 12,
          "overflow": "hidden",
          "shadowColor": "#000",
          "shadowOffset": {
            "height": 2,
            "width": 0,
          },
          "shadowOpacity": 0.1,
          "shadowRadius": 4,
        },
      ]
    }
  >
    <View
      style={
        {
          "position": "relative",
        }
      }
    >
      <View
        style={
          {
            "alignItems": "center",
            "aspectRatio": 0.6666666666666666,
            "backgroundColor": "#d1d1d1",
            "justifyContent": "center",
            "width": "100%",
          }
        }
      >
        <Text
          style={
            [
              {
                "color": "#11181C",
              },
              {
                "fontSize": 16,
                "lineHeight": 24,
              },
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
            ]
          }
        >
          No Image
        </Text>
      </View>
      <View
        style={
          {
            "bottom": 8,
            "position": "absolute",
            "right": 8,
          }
        }
      >
        <View
          style={
            [
              {
                "backgroundColor": "#fff",
              },
              {
                "backgroundColor": "#0a7ea4",
                "borderRadius": 16,
                "paddingHorizontal": 8,
                "paddingVertical": 4,
              },
            ]
          }
        >
          <Text
            style={
              [
                {
                  "color": "#11181C",
                },
                {
                  "fontSize": 16,
                  "lineHeight": 24,
                },
                undefined,
                undefined,
                undefined,
                undefined,
                {
                  "color": "white",
                  "fontSize": 12,
                  "fontWeight": "bold",
                },
              ]
            }
          >
            7.5
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        {
          "gap": 4,
          "padding": 12,
        }
      }
    >
      <Text
        numberOfLines={2}
        style={
          [
            {
              "color": "#11181C",
            },
            undefined,
            undefined,
            {
              "fontSize": 16,
              "fontWeight": "600",
              "lineHeight": 24,
            },
            undefined,
            undefined,
            undefined,
          ]
        }
      >
        Test Movie
      </Text>
      <Text
        style={
          [
            {
              "color": "#11181C",
            },
            {
              "fontSize": 16,
              "lineHeight": 24,
            },
            undefined,
            undefined,
            undefined,
            undefined,
            {
              "fontSize": 14,
              "opacity": 0.7,
            },
          ]
        }
      >
        January 1, 2023
      </Text>
    </View>
  </View>
</View>
`;
